scripts/exregional_process_bufr.sh

```diff
@@ -0,0 +1,382 @@
+#
+# Source the variable definitions file and the bash utility functions.
+#
+#-----------------------------------------------------------------------
+#
+. $USHdir/source_util_funcs.sh
+source_config_for_task "task_process_bufr" ${GLOBAL_VAR_DEFNS_FP}
+#
+#-----------------------------------------------------------------------
+#
+# Save current shell options (in a global array).  Then set new options
+# for this script/function.
+#
+#-----------------------------------------------------------------------
+#
+{ save_shell_opts; . $USHdir/preamble.sh; } > /dev/null 2>&1
+#
+#-----------------------------------------------------------------------
+#
+# Get the full path to the file in which this script/function is located 
+# (scrfunc_fp), the name of that file (scrfunc_fn), and the directory in
+# which the file is located (scrfunc_dir).
+#
+#-----------------------------------------------------------------------
+#
+scrfunc_fp=$( readlink -f "${BASH_SOURCE[0]}" )
+scrfunc_fn=$( basename "${scrfunc_fp}" )
+scrfunc_dir=$( dirname "${scrfunc_fp}" )
+#
+#-----------------------------------------------------------------------
+#
+# Print message indicating entry into script.
+#
+#-----------------------------------------------------------------------
+#
+print_info_msg "
+========================================================================
+Entering script:  \"${scrfunc_fn}\"
+In directory:     \"${scrfunc_dir}\"
+
+This is the ex-script for the task that runs bufr (cloud, metar, lightning) preprocess
+with FV3 for the specified cycle.
+========================================================================"
+#
+#-----------------------------------------------------------------------
+#
+# Extract from CDATE the starting year, month, day, and hour of the
+# forecast.  These are needed below for various operations.
+#
+#-----------------------------------------------------------------------
+#
+START_DATE=$(echo "${CDATE}" | sed 's/\([[:digit:]]\{2\}\)$/ \1/')
+YYYYMMDDHH=$(date +%Y%m%d%H -d "${START_DATE}")
+JJJ=$(date +%j -d "${START_DATE}")
+
+YYYY=${YYYYMMDDHH:0:4}
+MM=${YYYYMMDDHH:4:2}
+DD=${YYYYMMDDHH:6:2}
+HH=${YYYYMMDDHH:8:2}
+YYYYMMDD=${YYYYMMDDHH:0:8}
+
+YYJJJHH=$(date +"%y%j%H" -d "${START_DATE}")
+PREYYJJJHH=$(date +"%y%j%H" -d "${START_DATE} 1 hours ago")
+
+#
+#-----------------------------------------------------------------------
+#
+# Get into working directory
+#
+#-----------------------------------------------------------------------
+#
+print_info_msg "$VERBOSE" "
+Getting into working directory for BUFR observation process ..."
+
+cd_vrfy ${DATA}
+
+fixgriddir=$FIXgsi/${PREDEF_GRID_NAME}
+
+print_info_msg "$VERBOSE" "fixgriddir is $fixgriddir"
+
+#
+#-----------------------------------------------------------------------
+#
+# link or copy background files
+#
+#-----------------------------------------------------------------------
+
+cp_vrfy ${fixgriddir}/fv3_grid_spec          fv3sar_grid_spec.nc
```

Comment:
Not sure why there are so many spaces here

```suggestion
cp_vrfy ${fixgriddir}/fv3_grid_spec fv3sar_grid_spec.nc
```